presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -835,6 +869,34 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
         return clientBuilder.build();
     }
 
+    private InitiateMultipartUploadResult initMultipartUpload(String bucket, String key)
```

Comment:
Nit: move this method below, to be directly before `PrestoS3InputStream`. The methods in this section are related to credentials and creating the S3 client.